core/src/main/scala/Utils.scala

```diff
@@ -13,6 +16,21 @@ package object utils {
       case -\/(e) => Task.fail(conv(e))
       case \/-(a) => Task.now(a)
     }
+    def toProcess(implicit conv: E => Throwable): Process[Task, A] = a match {
+      case -\/(e) => Process.fail(conv(e))
+      case \/-(a) => Process.emit(a)
+    }
+  }
+
+  implicit class AugmentedTask[A](t: Task[A]) {
+    def onComplete(f: Throwable \/ A => Unit): Task[A] =
+      t.attempt.flatMap{a => f(a); t}
+  }
+
+  implicit class AugmentedProcess[A](p: Process[Task, A]) {
+    def flatten[B](implicit conv: A => Process[Task,B]): Process[Task,B] =
+      p.flatMap(conv)
+    def uncons: Task[(A, Process[Task, A])] = (p pipe Process.await1).runLastOr(throw new NoSuchElementException).map(a => (a, p))
```

Comment:
@runarorama Can you review `uncons`. Could not come up with a better implementation than this (but it works)